src/main/java/com/taosdata/jdbc/TSDBResultSetBlockData.java

```diff
@@ -259,6 +279,23 @@ public void setByteArray(byte[] value) {
             buffer.position(pHeader);
             colData.add(col);
         }
+        semaphore.release();
+    }
+
+    public void doneWithNoData(){
+        semaphore.release();
+    }
+
+    public void waitTillOK() throws SQLException {
+        try {
+            // must be ok When the CPU has idle time
+            if (!semaphore.tryAcquire(5, TimeUnit.SECONDS))
```

Comment:
Is 5 seconds enough?

Reply:
yes, wait for computation in doSetByteArray, it will be done in a little ms